libraries/chain/include/graphene/chain/account_object.hpp

```diff
@@ -96,6 +96,29 @@ namespace graphene { namespace chain {
           */
          share_type pending_vested_fees;
 
+         /**
+          * Specifies the total voting weight of this account (set in maintenance interval)
+          */
+         struct voting_stat
+         {
+            bool has_proxy;
+            uint64_t self_voting_power;
+            flat_map<account_id_type, uint64_t> proxy_for;
+
+            const uint64_t get_total_voting_power() const
```

Comment:
brauchts total?

self_voting_power und voting_power passt eigtl ganz gut

Reply:
Also `self` ist die Power die man selbst hat.
`total` ist die man effektive hat. Wenn ein Proxy gesetzt ist, hat man keine effektive Power die wird ja dann dem Proxy gegeben.

Reply:
@sschiessl-bcp 

> Allgemeine Frage:
> 
> Wird das account_statistic object immer nur währent maintenance geupdated?
> Und falls ja, weiß das statistics object an welchem maintenance interval es geändert wurde? bzw wann der letzte war?

Account_stat im allgemeinen wird nich nur während maint geupdated. Glaub das keept auch track über lifetime operations gesendet etc. Deshalb hat es auch kein Reiter für in welchem maint es geändert wurde. 

Man könnte aber speziell für den Fall das `voting_stat` verändern und dort das maintenance interval zusätzlich hinterlegen.

Reply:
Ich meinte `get_total_voting_power` könnte doch einfach nur `get_voting_power` heißen. Beauty mark, ich denk es passt schon dann denkt man nochmal drüber nach was `total` bedeutet.

Kannst du das im Header der Klasse beschreiben?

Reply:
Ich würde es gern garnicht erst `power` sondern `stake` nennen. Einfach um konsistent mit dem Rest des codes zu bleiben

Reply:
> Kannst du das im Header der Klasse beschreiben?

Ich machn Kommentar rein..

> Das wär nice, vielleicht einfach die block_num in der es erzeugt worden ist

Alles klar mach ich mit rein.

Reply:
> > 
> 
> Das wär nice, vielleicht einfach die block_num in der es erzeugt worden ist

Je mehr daten du in das objekt reinschreibst, desto mehr RAM brauchen wir und desto eher werden die core devs nach dem kosten/Nutzen Fragen.

Ich denke voting statistics sollten nicht teil des kerns sein, sondern eher teil eines plugins oder einer datenbank werden.

Voting Stake soll eigentlich nur deshalbt hier rein, damit man ne möglichkeit hat die Chain nach essentiellen Governance parametern zu fragen

Reply:
> Ich denke voting statistics sollten nicht teil des kerns sein, sondern eher teil eines plugins oder einer datenbank werden.

Stimm ich voll und ganz zu. Das object in `account statistics` zu speichern ist nur eine der Möglichkeiten. Die Zuordnung zu einem block wäre wichtig für Historie, ist aber in dem `current state` überflüssig (da dort immer die zuordning "letztes Maintenance" gilt).

Wenn man das object dann in ES (oder sonst wo speichert) zusätzlich zum current state, dann bräuchte es block_num und account_id

Reply:
> Ich würde es gern garnicht erst `power` sondern `stake` nennen. Einfach um konsistent mit dem Rest des codes zu bleiben

Okay ich nenns um.

> Je mehr daten du in das objekt reinschreibst, desto mehr RAM brauchen wir und desto eher werden die core devs nach dem kosten/Nutzen Fragen.

Ich schau mal nach einer anderen Möglichkeit. Vielleicht findet sich ja was wie man rausfindet wann die letzte maintenance war.